app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :email, presence: true, length: { maximum: 255 },
+  format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
+
+  has_secure_password
```

Comment:
has_secure_password dùng để làm gì?

Reply:
Dạ, has_secure_password là một built-in method, cho phép 
- lưu mật khẩu dưới dạng băm (Hash được thực hiện nhờ attr password_digest.)
- thực hiện confirm password.
- trả về user khi password đúng ạ.